src/wast-parser.cc

```diff
@@ -355,6 +355,19 @@ void ResolveFuncTypes(Module* module) {
   }
 }
 
+void MarkDeclaredFunctions(Module* module) {
```

Comment:
This logic should be in the validator, not the parser/binary-reader

Reply:
One problem with doing that is the validator generally doesn't mutate the IR (it takes a const Module).

I did consider doing this internally withing the validator using a side data structure rather then embedding this `is_declared` field in the IR itself.   Do you think that would be better approach?  I'll give that a go and see what you think..

Reply:
Yeah, I'd prefer it that way I think. Storing it in the IR seems odd to me.